compiler/next/include/chpl/uast/Local.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_LOCAL_H
+#define CHPL_UAST_LOCAL_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/uast/Expression.h"
+
+namespace chpl {
+namespace uast {
+
+
+/**
+  This class represents a local statement.
+ */
+class Local final : public Expression {
+ private:
+  Local(ASTList stmts, bool exprExists, bool usesDo);
+  bool contentsMatchInner(const ASTNode* other) const override;
+  void markUniqueStringsInner(Context* context) const override;
+  bool exprExists_;
+  bool usesDo_;
+
+ public:
+
+  /**
+    Create and return a local statement rooted on the given expression and
```

Comment:
Let's use different terms like "with conditional" rather than "rooted on". Here and in a few other places in this PR.